index.d.ts

```diff
@@ -2207,6 +2333,94 @@ declare namespace Eris {
     toJSON(props?: string[]): JSONCache;
   }
 
+  //Interaction
+  export class Interaction extends Base {
+    applicationID: string;
+    id: string;
+    token: string;
+    type: InteractionType | number;
+    version: number;
+  }
+
+  export class PingInteraction extends Interaction {
+    type: 1;
+    acknowledge(): Promise<void>;
+    pong(): Promise<void>;
+  }
+
+  export class CommandInteraction extends Interaction {
+    channelID: string;
+    data: {
+      id: string;
+      name: string;
+      resolved?: {
+        users?: {[id: string]: User};
+        members?: { [id: string]: Omit<Member, "user" | "deaf" | "mute"> };
+        roles?: { [id: string]: Role };
+        channels?: { [id: string]: PartialChannel };
+      };
+      options?: InteractionDataOptions[];
+    };
+    guildID?: string;
+    member?: Member;
+    type: 2;
+    user?: User;
+    acknowledge(flags?: number): Promise<void>;
+    createFollowup(content: string | InteractionWebhookContent): Promise<Message>;
+    createMessage(content: string | InteractionContent): Promise<void>;
+    defer(flags?: number): Promise<void>;
+    deleteMessage(messageID: string): Promise<void>;
+    deleteOriginalMessage(): Promise<void>;
+    editMessage(messageID: string, content: string | MessageWebhookContent): Promise<Message>;
+    editOriginalMessage(content: string | MessageWebhookContent): Promise<Message>;
+    getOriginalMessage(): Promise<Message>
+  }
+
+  export class ComponentInteraction extends Interaction {
+    channelID: string;
+    data: {
+      componentType: 2 | 3;
+      custom_id: string;
+      values?: string[];
```

Comment:
it might be worth making button & select menu interactions distinct *in types only*, so we don't have to check if this exists, componentType can be used to narrow (though this might be more of a mess than it's worth)

Reply:
yeaa I honestly think it is, and exists with select menus so just check the component type?

Reply:
Cool